codalab/worker/local_run/local_dependency_manager.py

```diff
@@ -13,11 +13,12 @@
 import codalab.worker.pyjson
 from codalab.worker.worker_thread import ThreadDict
 from codalab.worker.state_committer import JsonStateCommitter
+from codalab.worker.bundle_state import DependencyKey
 
 logger = logging.getLogger(__name__)
 
 DependencyState = namedtuple(
-    'DependencyState', 'stage dependency path size_bytes dependents last_used message killed'
+    'DependencyState', 'stage dependency_key path size_bytes dependents last_used message killed'
```

Comment:
This will be a breaking change for the workers as well.
During next release we should bring down all workers, clear all their caches and bring them up from scratch.

Reply:
If we're going to do this, can we put in some mechanism to make workers more robust to changes in version and fail gracefully?  Want to minimize disruption.  Also, somewhere we should make a note of these breaking changes so that the next person who deploys is aware.  They are probably not going to see this comment.

Reply:
Yeah, that's a good thing to add. Can someone else from the Codalab team pick up the task (see if the server already responds with the version number to worker checkin and if so get the worker to log a warning that it's out of date). Again I can guide with that. Of course it won't help people with old workers who don't have the warning code, but hopefully there aren't too many personal cl-workers out there.

As for release notes, it may be a good idea to make a PR template like we do at SM, and then release people can just go over the PR notes of all the commits merged since last release to compile this stuff.